src/util/text.ts

```diff
@@ -250,182 +255,406 @@ export function isAlphaOrDecimal(c: i32): bool {
   return isAlpha(c) || isDecimal(c);
 }
 
-/** Tests if the specified character code is a valid start of an identifier. */
-export function isIdentifierStart(c: i32): bool {
-  return isAlpha(c)
-      || c == CharCode._
-      || c == CharCode.DOLLAR
-      || c >= 170 && c <= 65500
-         && lookupInUnicodeMap(c as u16, unicodeIdentifierStart);
+/** Tests if the specified code point is a valid start of an identifier. */
+export function isIdentifierStart(cp: i32): bool {
+  return isAlpha(cp)
+      || cp == CharCode._
+      || cp == CharCode.DOLLAR
+      || cp >= 170 && cp <= unicodeIdentifierStart[unicodeIdentifierStart.length - 1]
```

Comment:
How about cache `unicodeIdentifierStart[unicodeIdentifierStart.length - 1]` in global constant?